x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -575,6 +575,15 @@ private PhysicalOperation planLookupJoin(LookupJoinExec join, LocalExecutionPlan
         if (localSourceExec.indexMode() != IndexMode.LOOKUP) {
             throw new IllegalArgumentException("can't plan [" + join + "]");
         }
+        Map<String, IndexMode> indicesWithModes = localSourceExec.index().indexNameWithModes();
+        if (indicesWithModes.size() != 1) {
+            throw new IllegalArgumentException("can't plan [" + join + "], found more than 1 index");
```

Comment:
Since you added yaml tests, that is also a convenient place to write negative tests that assert we get this error.

Reply:
This is checked in the Analyzer (ResolveTable), so it should be "impossible" to reach this point I think.
I added the explicit throw here anyway as a guard, because it's required here. Could make it an `assert`, but there are other cases in this file with similar throws, so I tried to be consistent